packages/wrangler/src/generate.ts

```diff
@@ -129,3 +134,159 @@ function generateWorkerDirectoryName(workerName: string): string {
+ * and is missing support for:
+ * - `http`
+ * - `ftp(s)`
+ * - `file`
+ * - `ssh`
+ */
+const TEMPLATE_REGEX =
+	/^(?:(?:https:\/\/)?(?<httpsUrl>[^:/]+\.[^:/]+)\/|git@(?<gitUrl>[^:/]+)[:/]|(?<shorthandUrl>[^/]+):)?(?<user>[^/\s]+)\/(?<repository>[^/\s#]+)(?:(?<subdirectoryPath>(?:\/[^/\s#]+)+))?(?:\/)?(?:#(?<tag>.+))?/;
+
+// there are a few URL formats we support:
+// - `user/repo` -> assume github, use "https://github.com/user/repo.git"
+// - `https://<httpsUrl>
+// - `git@<gitUrl>`
+// - `(bb|bitbucket|gh|github|gl|gitlab):user/repo` -> parse shorthand into https url
+
+/**
+ * There's no URL, so assume a github repo
+ */
+type NoUrl = {
+	httpsUrl: undefined;
+	gitUrl: undefined;
+	shorthandUrl: undefined;
+};
+
+/**
+ * A https url (e.g. https://bitbucket.org/user/repo)
+ */
+type HttpsUrl = Omit<NoUrl, "httpsUrl"> & { httpsUrl: string };
+
+/**
+ * A git url (e.g. git@gitlab.com:user/repo)
+ */
+type GitUrl = Omit<NoUrl, "gitUrl"> & { gitUrl: string };
+
+/**
+ * A shorthand url (e.g. github:user/repo)
+ */
+type ShorthandUrl = Omit<NoUrl, "shorthandUrl"> & { shorthandUrl: string };
+
+/**
+ * Union of all possible URL groups. Exactly one will be present, the rest
+ * will be `undefined`.
+ */
+type TemplateRegexUrlGroup = NoUrl | HttpsUrl | GitUrl | ShorthandUrl;
+
+/**
+ * Possible matches of `TEMPLATE_REGEX` against a passed-in template arg
+ */
+type TemplateRegexGroups = {
+	/** The user the repo is under */
+	user: string;
+
+	/** The repo name */
+	repository: string;
+
+	/** Optional, path to subdirectory containing template. Begins with `/` */
+	subdirectoryPath?: string;
+
+	/** Optional tag (or branch, etc.) to clone */
+	tag?: string;
+} & TemplateRegexUrlGroup;
+
+/**
+ * Parses a regex match on any of the URL groups into a URL base
+ *
+ * @param urlGroup a regex hit for a URL of any sort
+ * @returns the protocol and domain name of the url to clone from
+ */
+function toUrlBase({ httpsUrl, gitUrl, shorthandUrl }: TemplateRegexUrlGroup) {
+	if (httpsUrl !== undefined) {
+		return `https://${httpsUrl}`;
+	}
+
+	if (gitUrl !== undefined) {
+		return `git@${gitUrl}`;
+	}
+
+	if (shorthandUrl !== undefined) {
+		switch (shorthandUrl) {
+			case "github":
+			case "gh":
+				return "https://github.com";
+			case "gitlab":
+			case "gl":
+				return "https://gitlab.com";
+			case "bitbucket":
+			case "bb":
+				return "https://bitbucket.org";
+			default:
+				throw new Error(
+					`Unable to parse shorthand ${shorthandUrl}. Supported options are "bitbucket" ("bb"), "github" ("gh"), and "gitlab" ("gl")`
+				);
+		}
+	}
+
+	return "https://github.com";
+}
+
+/**
+ * Parses a template string (e.g. "user/repo", "github:user/repo/path/to/subdirectory")
+ * into a remote URL to clone from and an optional subdirectory to filter for
+ *
+ * @param template the template string to parse
+ * @returns an object containing the remote url and an optional subdirectory to clone
+ */
+function parseTemplate(template: string): {
```

Comment:
NIT: I was thinking this function actually parsed the contents of a template. But it is actually just the "template path". How about:

```suggestion
function parseTemplatePath(templatePath: string): {
```